main.py

```diff
@@ -77,28 +78,23 @@ def root():
 
 
 
-@app.route('/adopt/<listing_id>', methods=['GET', 'POST'])
+@app.route('/adopt/<listing_id>', methods=['POST'])
 def adopt(listing_id):
-    form = AdoptionForm()
-        
     listing = get_listing(listing_id)
-    if not listing:
-        app.logger.error(f"Failed to get listing: {listing_id}")
-        abort(404, description=f"Failed to get listing: {listing_id}")
-
-    if form.validate_on_submit():
```

Comment:
Why is form validation ignored?

Reply:
Because now the form is submitted directly from root page, '/' and create a POST request to /adopt/<listing_id>, so request.form comes in the form of a immutable dictionary. No form validation can be made at the /adopt/<listing_id> side. But there is still native frontend form validation through Jinja.